### PR TITLE
chore(ci.jenkins.io): add a AWS windows 2022 agent to ci.jenkins.io

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3336

### CODE REVIEW

File: hieradata/clients/azure.ci.jenkins.io.yaml [hunk 1]
```diff
@@ -352,6 +352,22 @@ profile::jenkinscontroller::jcasc:
...
+            maxInstances: 10
+            instanceType: T3Xlarge # 4 vCPUs / 16 Gb
+            os: "windows"
+            os_version: "2022"
+            architecture: "amd64"
+            labels:
+              - docker-windows-2022
+              - windows-2022
+            useAsMuchAsPossible: true
+            spot: true
```
Comment (lemeurherve) [inline hieradata/clients/azure.ci.jenkins.io.yaml]: Not directly related to this PR, do we really want "spot" machine for Windows?
Windows builds are usually long, not sure if taking the risk of getting them interrupted at any time is the right choice.

**Review by lemeurherve — DISMISSED** [review]

File: hieradata/clients/azure.ci.jenkins.io.yaml (hunk 1 above)
Comment (dduportal) [reply hieradata/clients/azure.ci.jenkins.io.yaml]: Good point. It won't kill to switch them, like we did for highmem VM agents, to "normal" workload (more expansive of course).

@lemeurherve can you open a PR for setting ALL windows templates to be non spot? Don't forget this PR :)

**Review by dduportal — DISMISSED** [review]
@smerle33 Approving: you'll have to manage the merge of this PR yourself so you can control the puppet deployment without any surprise

File: hieradata/vagrant/common.yaml [hunk 2]
```diff
@@ -226,6 +226,23 @@ profile::jenkinscontroller::jcasc:
             architecture: "amd64"
             labels:
               - docker-windows
+              - docker-windows-2019
+            useAsMuchAsPossible: true
+            spot: true
```
Comment (lemeurherve) [inline hieradata/vagrant/common.yaml]:
```suggestion
            spot: false
```

Comment (lemeurherve) [reply hieradata/vagrant/common.yaml]: Preparing the PR to remove it from the other Windows agents, no need to include it in this one.

File: hieradata/clients/azure.ci.jenkins.io.yaml (hunk 1 above)
Comment (lemeurherve) [inline hieradata/clients/azure.ci.jenkins.io.yaml]:
```suggestion
            spot: false
```

File: hieradata/vagrant/common.yaml (hunk 2 above)
Comment (lemeurherve) [reply hieradata/vagrant/common.yaml]: See https://github.com/jenkins-infra/jenkins-infra/pull/2585

Comment (dduportal) [reply hieradata/vagrant/common.yaml]: The other PR is not merged. There is no reason to link the 2 of them.

Comment (lemeurherve) [reply hieradata/vagrant/common.yaml]: Yes sure, I'm just asking to not include `spot: true` here as this one isn't merged yet too.

Comment (lemeurherve) [reply hieradata/vagrant/common.yaml]: I don't mind doing a little bit more work on my other PR after this one is merged, it just seemed to me easier like that.

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**